extensions/data-transfer/portability-data-transfer-twitter/src/main/java/org/dataportabilityproject.transfer.twitter/TwitterTransferExtension.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.transfer.twitter;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.List;
+import org.dataportabilityproject.api.launcher.ExtensionContext;
+import org.dataportabilityproject.spi.cloud.storage.AppCredentialStore;
+import org.dataportabilityproject.spi.transfer.extension.TransferExtension;
+import org.dataportabilityproject.spi.transfer.provider.Exporter;
+import org.dataportabilityproject.spi.transfer.provider.Importer;
+import org.dataportabilityproject.types.transfer.auth.AppCredentials;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TwitterTransferExtension implements TransferExtension {
+  private static final List<String> SUPPORTED_TYPES = ImmutableList.of("photos");
+  private static final Logger LOGGER = LoggerFactory.getLogger(TwitterTransferExtension.class);
+  private static final String TWITTER_KEY = "TWITTER_KEY";
```

Comment:
could these be defined in a central place since they are also defined in the auth extension? Not sure if we do that for other extensions, so I'm fine leaving it alone too

Reply:
I think we should move these key values out of the plugins in general and have the framework only send the right values  on to the plugins.  Right now Twitter gets to see all the other plugin's vlaues, which isn't ideal.